Contest_API.md

```diff
@@ -621,13 +621,39 @@ The following endpoints are associated with languages:
+| -------------------- | ------ | --------- | --------- | ---------- | --------------------------------------------------------------------- |
+| id                   | ID     | yes       | no        | CCS        | identifier of the language from table below                           |
+| name                 | string | yes       | no        | CCS        | name of the language (might not match table below, e.g. if localized) |
+| compiler             | Command object | no | yes      | CCS        | Command used for compiling submissions. |
+| runner               | Command object | no | yes      | CCS        | Command used for running submissions. Relevant e.g. for interpreted languages and languages running on a VM. |
 | entry_point_required | boolean         | yes       | no        | CCS        | whether the language requires an entry point                                                                            |
 | entry_point_name     | string          | depends   | yes       | CCS        | the name of the type of entry point, such as "Main class" or "Main file"). Required iff entry_point_required is present |
 | extensions           | array of string | yes       | no        | CCS        | file extensions for the language                                                                                        |
+| compiler             | Command object  | no        | yes       | CCS        | Command used for compiling submissions.                                                                                 |
+| runner               | Command object  | no        | yes       | CCS        | Command used for running submissions. Relevant e.g. for interpreted languages and languages running on a VM.            |
+
+JSON elements of Command objects:
+
+| Name            | Type   | Required | Description |
+| :-------------- | :----- | :------- | :---------- |
+| command         | string | yes      | Command to run.                                                                        |
+| args            | string | no       | Argument list for command. {files} denotes where to include the file list.             |
+| version         | string | no       | Expected output from running the version-command.                                      |
+| version-command | string | no       | Command to run to get the version. Defaults to `<command> --version` if not specified. |
+
+The compiler and runner elements are intended for informational purposes. It is not expected that systems will synchronize compiler and runner settings via this interface.
+
+JSON elements of Command objects:
+
+| Name            | Type   | Required | Description |
+| :-------------- | :----- | :------- | :---------- |
+| command         | string | yes      | Command to run.                                              |
+| args            | string | no       | Argument list for command. `{files}` denotes where to include the file list. |
```

Comment:
Late to the party, but we need other vars similar `{files}`, e.g. for derivatives of the entry point (think main class in Java and Kotlin).

Reply:
Also for Java the heap size is dependent on the problem's memory limit, so we would need a way to express that.

Reply:
This issue was merged a year ago, so I'm worried that comments here might be forgotten. I'd open up an issue, submit a PR with a suggestion (even if not complete, it'll generate discussion) or join an upcoming call.

Reply:
I promised Tobi to put this in the agenda, which I just did.

Reply:
> Late to the party, but we need other vars similar `{files}`,

Yes. agreed.